core/src/main/java/org/bitcoinj/base/AddressParser.java

```diff
@@ -17,42 +17,93 @@
-     * but is used in a context where a specific {@link Network} has been specified. This interface may be
-     * implemented by creating a partial application of ({@link AddressParser#parseAddress(String, Network)} providing
-     * a fixed value for {@link Network}.
+     * Interface implemented by custom address parser providers.
      */
-    @FunctionalInterface
-    interface Strict {
+    interface AddressParserProvider {
+        /**
+         * Return a parser that will parse valid addresses for all networks ({@link Network}) known by this provider.
+         * @return a parser for all networks known by this provider
+         */
+        AddressParser forKnownNetworks();
+
         /**
-         * Parse an address in a strict context (e.g. the network must be valid)
-         * @param addressString string representation of address
-         * @return A validated address object
-         * @throws AddressFormatException invalid address string or not valid for network (provided by context)
+         * Return a parser that will parse valid addresses for a given {@link Network}.
+         * @param network network to parse and validate addresses for
+         * @return a parser for the specified network
          */
-        Address parseAddress(String addressString) throws AddressFormatException;
+        AddressParser forNetwork(Network network);
+    }
+
+    /**
+     * Get a <i>legacy</i> address parser that knows about networks that have been
+     * dynamically added to the list maintained by {@link org.bitcoinj.params.Networks}.
+     * @return A parser for all known networks
+     */
+    @Deprecated
+    static AddressParser getLegacy() {
+        return DefaultAddressParserProvider.fromNetworks().forKnownNetworks();
+    }
+
+    /**
+     * Get a <i>legacy</i> address parser that knows about networks that have been
+     * dynamically added to the list maintained by {@link org.bitcoinj.params.Networks}.
+     * @param network the network to parse for
+     * @return A parser that will throw for strings that are not valid for network.
+     */
+    @Deprecated
+    static AddressParser getLegacy(Network network) {
+        return DefaultAddressParserProvider.fromNetworks().forNetwork(network);
+    }
+
+    /**
+     * Get a <i>legacy</i> address parser that knows about networks that have been
+     * dynamically added to the list maintained by {@link org.bitcoinj.params.Networks}.
+     * @param params the network to parser for, or {@code null} for all networks.
+     * @return A parser that will throw for strings that are not valid for network.
+     */
+    @Deprecated
+    static AddressParser getLegacy(@Nullable NetworkParameters params) {
+        AddressParser.AddressParserProvider provider = DefaultAddressParserProvider.fromNetworks();
+        return (params == null)
+                ? provider.forKnownNetworks()
+                : provider.forNetwork(params.network());
     }
+
```

Comment:
Would you place a different type of command here if you needed to? Such as if you needed to make changes or additions to a different type of command?